tests/sps/validation/test_peer_review.py

```diff
@@ -824,14 +824,15 @@ def test_peer_review_keys_validation(self):
             link_type_list=['doi']
         )
 
-        response_dicts = list(validations.nodes_validation)
+        obtained_dicts = list(validations.nodes_validation)
 
-        expected_keys = {'title', 'parent', 'parent_id', 'item', 'sub_item', 'validation_type', 'response',
-                         'expected_value', 'got_value', 'message', 'advice'}
+        expected_keys = ['title', 'parent', 'parent_id', 'item', 'sub_item', 'validation_type', 'response',
+                         'expected_value', 'got_value', 'message', 'advice']
 
-        for i, response_dict in enumerate(response_dicts):
-            with self.subTest(i):
-                self.assertSetEqual(set(response_dict.keys()), expected_keys)
+        for expected_key in expected_keys:
+            for obtained_dict in obtained_dicts:
+                with self.subTest(expected_key):
```

Comment:
Dentro de subTest vc tem que usar um identificador para o teste, pois se ele falhar vai indicar o "id" do teste. Como o expected_key vc considerou como id, percebe que não é adequado porque está dentro de um for (do obtained_dicts). Consulte self.subTest para melhor escolher o "id", acho que ele achei mais de um parâmetro

Reply:
Outra opção é voltar ao que estava antes